test/LONStaking/RageExit.t.sol

```diff
@@ -100,14 +98,14 @@ contract TestLONStakingRageExit is TestLONStaking {
         _rageExitAndValidate(user);
     }
 
-    function testFuzz_RageExitOneByOnePlusPenaltyWithMultipleStake(uint80[16] memory stakeAmounts) public {
-        // LON cap lies between 2**87 and 2**88, setting upper bound of stake amount to 2**80 - 1 so stake amount will not exceed cap
+    function testFuzz_RageExitOneByOnePlusPenaltyWithMultipleStake(uint256[16] memory stakeAmounts) public {
         uint256 totalLONAmount = lon.totalSupply();
         for (uint256 i = 0; i < stakeAmounts.length; i++) {
-            uint256 stakeAmount = stakeAmounts[i];
-            vm.assume(stakeAmount > 0);
-            totalLONAmount = totalLONAmount.add(stakeAmount);
-            vm.assume(totalLONAmount <= lon.cap());
+            uint256 itemsLeft = stakeAmounts.length - i - 1;
+            // If stake amount is set to `lon.cap().sub(totalLONAmount)`, rest of the stake amount will all be zero and become invalid.
+            // So an additional `itemsLeft` is subtracted from the max value of the current stake amount
+            stakeAmounts[i] = bound(stakeAmounts[i], 1, lon.cap().sub(totalLONAmount).sub(itemsLeft));
+            totalLONAmount = totalLONAmount.add(stakeAmounts[i]);
```

Comment:
I am still not getting the reason to subtract additional number, especially it is an array index but subtraction from amount, which looks weird to me. 🤔 Could you provide a detailed example for it?

Reply:
Another thing is that these comments and actions seem to spread across multiple files which might incur maintaining problem in the future. It would be better if it could maybe be collected into an util to share among cases.

Reply:
We need the stake amount to greater than `1` (and the sum of all the stake amount to be less than `lon.cap()`). So when bounding stake amount in each iteration, we need to make sure that rest of the stake amounts can still be greater than `1` and that's why the max stake amount is `lon.cap().sub(totalLONAmount).sub(itemsLeft)`. `.sub(itemsLeft)` is there to make sure rest of the stake amount can be greater than `1` (if those stake amounts are all `1`, their sum is `itemsLeft`).

Reply:
Maybe I should rename `itemsLeft` to `numStakeAmountLeft` and rewrite stake amount bound as `stakeAmounts[i] = bound(stakeAmounts[i], MIN_STAKE_AMOUNT, lon.cap().sub(totalLONAmount).sub(numStakeAmountLeft  * MIN_STAKE_AMOUNT));`

Reply:
Thanks for the explanation, it's much clearer 👍 I think the new suggestion you gave is more meaningful to me. We can go for it.

Reply:
Fixed in 1142793

Reply:
I moved the repeated bound function into util functions in da94bdf